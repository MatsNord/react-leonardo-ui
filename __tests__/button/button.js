import React from 'react';
import renderer from 'react-test-renderer';

import Button from '../../src/button';

test('Normal button renders correctly', () => {
  const tree = renderer.create(
    <Button>Text</Button>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button with subcomponents renders correctly', () => {
  const tree = renderer.create(
    <Button>
      <Button.Icon name="more" />
      <Button.Text>Text</Button.Text>
      <Button.Caret />
    </Button>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button with states renders correctly', () => {
  const tree = renderer.create(
    <Button value="Value" active disabled />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button with modifiers renders correctly', () => {
  const tree = renderer.create(
    <Button value="Value" variant="success" size="large" block rounded />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button passes props correctly', () => {
  const onClick = jest.fn(() => {});

  const tree = renderer.create(
    <Button className="my-class" data-x="x" title="Title" onClick={onClick}>
      Text
    </Button>
  ).toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.onClick();
  expect(onClick.mock.calls.length).toBe(1);
});
