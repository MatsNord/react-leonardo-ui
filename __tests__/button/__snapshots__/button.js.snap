exports[`test Button passes props correctly 1`] = `
<button
  className="lui-button my-class"
  data-x="x"
  onClick={[Function]}
  title="Title">
  Text
</button>
`;

exports[`test Button with modifiers renders correctly 1`] = `
<button
  className="lui-button lui-button--success lui-button--large lui-button--block lui-button--rounded"
  value="Value" />
`;

exports[`test Button with states renders correctly 1`] = `
<button
  className="lui-button lui-active lui-disabled"
  value="Value" />
`;

exports[`test Button with subcomponents renders correctly 1`] = `
<button
  className="lui-button">
  <span
    aria-hidden="true"
    className="lui-icon lui-icon--more lui-button__icon" />
  <span
    className="lui-button__text">
    Text
  </span>
  <span
    className="lui-button__caret  lui-caret" />
</button>
`;

exports[`test Normal button renders correctly 1`] = `
<button
  className="lui-button">
  Text
</button>
`;
